src/librustc_typeck/check/demand.rs

```diff
@@ -415,10 +415,58 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                           src,
                                           if needs_paren { ")" } else { "" },
                                           expected_ty);
-            let into_suggestion = format!("{}{}{}.into()",
-                                          if needs_paren { "(" } else { "" },
-                                          src,
-                                          if needs_paren { ")" } else { "" });
+            let into_suggestion = format!(
+                "{}{}{}.into()",
+                if needs_paren { "(" } else { "" },
+                src,
+                if needs_paren { ")" } else { "" },
+            );
+            let literal_is_ty_suffixed = |expr: &hir::Expr| {
+                if let hir::ExprKind::Lit(lit) = &expr.node {
+                    lit.node.is_suffixed()
+                } else {
+                    false
+                }
+            };
+
+            let into_sugg = into_suggestion.clone();
+            let suggest_to_change_suffix_or_into = |err: &mut DiagnosticBuilder<'tcx>,
+                                                    expr: &hir::Expr,
+                                                    expected_ty: Ty<'tcx>,
+                                                    checked_ty: Ty<'tcx>,
+                                                    note: Option<&str>| {
```

Comment:
It's the first time I've seen a closure that needed multiple arguments and I'm surprised `rustfmt` suggests exactly this :)

Reply:
oh, `expr` and `*_ty` aren't needed actually, I'll remove them.